webapp/channels/src/components/user_settings/plugin/types.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export type PluginConfiguration = {
+    id: string;
+    uiName: string;
+    icon?: string;
+    settings: PluginConfigurationSetting[];
+}
+
+type BasePluginConfigurationSetting = {
+    name: string;
+    title: string;
+    helpText?: string;
+    onSubmit?: (name: string, value: string) => void;
+    default: string;
+}
+
+type PluginConfigurationRadioSetting = BasePluginConfigurationSetting & {
+    type: 'radio';
```

Comment:
This looks great! I know this is the only possible value for `PluginConfigurationSetting` currently, but I always like seeing this pattern since it makes the TS work greatt without us having to write a bunch of custom type guards. I look forward to finding other places to use it

Reply:
I have to admit this is "stolen" from the system console refactor done not long ago. But yeah, the pattern is really nice :smile: